### PR TITLE
fix: only request push notifications once per login session

### PR DESCRIPTION
**Pull request by brainbicycle**
This PR resolves [MOPLAT-737] 

### Description

Adds a check to make sure we only request push permissions once per login session to not spam Android users with the settings dialog when they deny push permissions. 

**Follow-ups:**
Digging into this code revealed it is fairly buggy, probably at least partially my fault but having the logic split across native side and typescript side makes it hard to follow. Going to create a ticket to bring this all over to typescript, write some tests and hopefully fix some of these bugs:
- We are not refreshing tokens when the users push status is authorized
- Login is not the relevant event for our push logic, we should be checking it any time the app is launched
- Not a bug necessarily but now that Android has a similar push permissions model we should make the logic similar

### PR Checklist

- [x] I have tested my changes on **iOS** and **Android**.
- [x] I hid my changes behind a **[feature flag]**, or they don't need one.
- [x] I have included **screenshots** or **videos**, or I have not changed the UI.
- [x] I have added **tests**, or my changes don't require any.
- [x] I added an **[app state migration]**, or my changes do not require one.
- [x] I have documented any **follow-up work** that this PR will require, or it does not require any.
- [x] I have added a **changelog entry** below, or my changes do not require one.

### To the reviewers 👀

- [x] I would like **at least one** of the reviewers to **run** this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

-

#### iOS user-facing changes

-

#### Android user-facing changes

- Fix push settings dialog showing many times - Brian 

#### Dev changes

-

</details>

Need help with something? Have a look at our [docs], or get in touch with us.

[app state migration]: ../blob/main/docs/adding_state_migrations.md
[feature flag]: ../blob/main/docs/developing_a_feature.md
[docs]: ../blob/main/docs/README.md

[MOPLAT-737]: https://artsyproduct.atlassian.net/browse/MOPLAT-737?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by damassi — APPROVED** [review]

File: ios/Artsy/App/ARAppNotificationsDelegate.m [hunk 1]
```diff
@@ -194,9 +194,9 @@ - (void)application:(UIApplication *)application didFailToRegisterForRemoteNotif
         @"action_name" : @"Cancel",
         @"context_screen"  : analyticsContext
     }];
-    [AROptions setBool:YES forOption:ARPushNotificationsAppleDialogueRejected];
 #if (TARGET_IPHONE_SIMULATOR == 0)
     ARErrorLog(@"Error registering for remote notifications: %@", error.localizedDescription);
+    [AROptions setBool:YES forOption:ARPushNotificationsAppleDialogueRejected];
```
Comment (brainbicycle) [inline ios/Artsy/App/ARAppNotificationsDelegate.m:199]: noticed that on simulator this will always fail so this bool gets set making the settings dialog show, which is confusing. Probably a good idea to test push on device always but moving this behind the check should make it a little less confusing.
